pkg/queryfrontend/config.go

```diff
@@ -242,14 +245,28 @@ type LabelsConfig struct {
 // Validate a fully initialized config.
 func (cfg *Config) Validate() error {
 	if cfg.QueryRangeConfig.ResultsCacheConfig != nil {
-		if cfg.QueryRangeConfig.SplitQueriesByInterval <= 0 {
-			return errors.New("split queries interval should be greater than 0 when caching is enabled")
+		if cfg.QueryRangeConfig.SplitQueriesByInterval <= 0 && cfg.QueryRangeConfig.QuerySplitThresholdInterval <= 0 {
+			return errors.New("split queries  or split threshold interval should be greater than 0 when caching is enabled")
 		}
 		if err := cfg.QueryRangeConfig.ResultsCacheConfig.Validate(querier.Config{}); err != nil {
 			return errors.Wrap(err, "invalid ResultsCache config for query_range tripperware")
 		}
 	}
 
+	if cfg.QueryRangeConfig.QuerySplitThresholdInterval != 0 && cfg.QueryRangeConfig.SplitQueriesByInterval != 0 {
```

Comment:
Should we rather check that either all flags are set or no flags are set? Users can still set `MaxQuerySplitInterval` without setting `QuerySplitThresholdInterval` which is not supported.

So something like
```
staticShardingSet := cfg.QueryRangeConfig.SplitQueriesByInterval != 0
dynamicShardingSet = ...

if staticShardingSet && dynamicShardingSet {
  return error
}
```